### ISSUE TITLE
[IPKernelApp] ERROR | UNKNOWN MESSAGE TYPE: 'comm_open'

### ISSUE THREAD

**Issue by scopatz** [ipython/ipykernel#64]
Hello all, I am writing a wrapper kernel for xonsh (see scopatz/xonsh#399). We noticed that the following error gets printed to the command line on jupyter v4.0.6, jupyter notebook v4.0.5, and ipykernel v4.0.3.  

```
[IPKernelApp] ERROR | UNKNOWN MESSAGE TYPE: 'comm_open'
```

Note that this error does not get printed if the jupyter notebook is on v4.0.4 and the other two stay the same.  Not really sure what is going on here. It doesn't seem to effect anything on Ubuntu 15.04, but may be effecting things on Windows.

**Comment by scopatz** [comment]
(That is, nothing is working on Windows right now...)

**Comment by minrk** [comment]
It's a little weird that a comm_open message is arriving, but wrapper kernels should safely ignore unhandled messages and move on. Does it work on 4.0.4, or also not work, and the only difference is the unknown message?

**Comment by scopatz** [comment]
It works just fine on v4.0.4.  The only difference is this unknown message, that does seem to be safely ignored.

**Comment by minrk** [comment]
And just upgrading/downgrading the notebook package in the same env is the only change? Bizarre, but at least that makes it a very small set of changes to look through. Are you using conda or pip to do this?

**Comment by scopatz** [comment]
Yes, the upgrade/downgrade was the only change. I used conda to do this.

**Comment by minrk** [comment]
And this is only an issue on Windows? Does an IPython kernel work in the same env? Does it matter what notebook you try to open?

I'm going with downright bonkers for now, but I'll see if I can dig out a comparison of the conda packages tomorrow. I don't see anything in the [repo comparison](https://github.com/jupyter/notebook/compare/4.0.4...4.0.5) that would be relevant, but there have been occasional issues with conda packages that get something amiss.

**Comment by scopatz** [comment]
hmmm yeah.  Nothing here seems to indicate this kind of error message.  

The issue for xonsh is on Windows, but may be unrelated. 

Don't loose too much sleep over this @minrk.

**Comment by adrian17** [comment]
I just noticed it too. Version 4.0.6. If it's not an actual issue, maybe the name should be changed to "warning"?

**Comment by evhub** [comment]
Same error is still appearing under 4.1.2.

**Comment by dimpase** [comment]
We also see it here: https://trac.sagemath.org/ticket/21416

**Comment by takluyver** [comment]
In general: you do not need to worry about this. The message type in question is optional for kernels to implement. I'll open a PR to make the log message less scary.

As an analogy, it's totally legitimate to host a website without a `favicon.ico`. But browsers will look for that, and your logs will show a 404 error from that. The other party is trying something, but is prepared to handle it not working.

**Comment by dimpase** [comment]
Thanks. Could you comment on the other `[IPKernelApp] ERROR` things there? Is it a buggy json given to the notebook, or something else?

**Comment by takluyver** [comment]
#230 downgrades it to a warning, and not SHOUTY CAPS.

**Comment by takluyver** [comment]
Those messages [come from the gap kernel](https://github.com/gap-packages/jupyter-kernel-gap/blob/eca704109f7ab4b64c1d522a3fd962749f918a77/jupyter_kernel_gap/kernel.py#L123), not from our code. I think somebody just pushed informational log messages up to error level to make them more visible.

**Comment by dimpase** [comment]
You might see  [here](https://github.com/gap-packages/jupyter-kernel-gap/blob/eca704109f7ab4b64c1d522a3fd962749f918a77/jupyter_kernel_gap/kernel.py#L53) that it's the only logging option they managed to dig up. Is there a better way?
